test/unit/median_oracle.js

```diff
@@ -77,7 +77,8 @@ contract('MedianOracle:pushReport', async function (accounts) {
     expect(await chain.isEthException(oracle.pushReport(1000000000000000000, { from: A }))).to.be.true;
     oracle.addProvider(A, { from: deployer });
     r = await oracle.pushReport(1000000000000000000, { from: A });
-    // should fail if reportDelaySec did not pass since the previous push
+    r = await oracle.pushReport(1000000000000000000, { from: A });
```

Comment:
can we create a separate test for this case?

I think the original test is `should only push from authorized source`

Reply:
Will be great if we can test all the cases:

Reported rates (ordered by time) are r1, r2. The new rate is r.

1) r2 < minValidTimestamp
2) r1 < minValidTimestamp < maxValidTimestamp < r2
3) r1 < minValidTimestamp < r2 < maxValidTimestamp
4) minValidTimestamp < r1 < r2 < maxValidTimestamp
5) minValidTimestamp < r1 < maxValidTimestamp < r2
6) minValidTimestamp < maxValidTimestamp < r1

Reply:
will make sure at least coverage is back to 100%